### DISCUSSION TITLE
Accept Terms on EC2 deployment

### DISCUSSION THREAD

**Discussion by cra-schleupen** [flatcar/Flatcar#1583] [Q&A]
I'm deploying Flatcar on EC2 with terraform. When I (re-)deploy the instance, I get an error message:

> Error: creating EC2 Instance: operation error EC2: RunInstances, https response error StatusCode: 401, RequestID:[...], api error OptInRequired: In order to use this AWS Marketplace product you need to accept terms and subscribe. To do so please visit https://aws.amazon.com/marketplace/pp?sku=...

I need to visit that page, accept the terms, wait 5 min and need to re-run terraform plan/apply.

This happens on first deploy into an AWS account and when I redeploy the instance after the latest version changes. I'm using the terraform code from the official [docs](https://www.flatcar.org/docs/latest/installing/cloud/aws-ec2/) (this doesn't include any comments about Terms/Subscription).

This renders the image usage impossible, especially since I need to accept those terms in every account. I can't imaging that is how the image is supposed to be used in production. 

I read some comments, like [this one](https://github.com/flatcar/Flatcar/issues/456#issuecomment-889311664) that they couldn't reproduce the issue. I already talked to our governance guys, they're not aware of any policy, that could enforce this behavior.

Can anyone give a hint about how to handle this?

**Comment by tormath1** [comment]
Hello @cra-schleupen thanks for opening this. Can you try to update this line:
https://github.com/flatcar/flatcar-terraform/blob/f2a215ceec284315915bdcd6a243ac298ccb263e/aws/aws-ec2-machines.tf#L103

with:
```tf
  owners      = ["075585003325"]
```

?

**Reply by cra-schleupen** [reply]
Using that owner it works without issues. But I'm not too thrilled about using a community AMI. Is there an option for Kinvolk to change their policy about the official Marketplace AMIs?

I mean, it literally says 

> Starting from $0.00 to $0.00/hr for software

**Reply by tormath1** [reply]
I am investigating on this license thing. In the meantime, note that the images from the marketplace and from the "community" account are the exact same Flatcar image under the hood and they are published at the same moment during the release.

**Reply by cra-schleupen** [reply]
An alternative approach could be to at least verify the provider of the community AMIs. 

For bottlerock AMIs the preview looks like this. Those are the AMIs uses by karpenter when bottlerocket is configured, so they seem to solve the licence issue the same way.
![image](https://github.com/user-attachments/assets/d2c9f127-731c-4e91-b9fe-c42d94731f0c)

Note the "Verified provider" batch.

While the flatcar there is no indication that the images are provided by you:
![image](https://github.com/user-attachments/assets/92f0aeca-9cc2-4f03-a1b3-de11ec410f8a)

If it would say "By Kinvolk" (maybe via OwnerAlias), as with the Marketplace AMI, the owner would be more trustworthy. Also it would be helpful if you could add the owner id to the official docs.
